hyperspy/signal.py

```diff
@@ -3111,6 +3112,79 @@ def crop(self, axis, start=None, end=None, convert_units=False):
         if convert_units:
             self.axes_manager.convert_units(axis)
 
+    def interpolate_on_axis(self,
+                            new_axis,
+                            index_in_axes_manager=0,
+                            inplace=False,
+                            set_navigate=True,
+                            degree=1):
+        """Replaces the axis specified by index_in_axes_manager with the provided new_axis
+        and interpolates data accordingly.
+
```

Comment:
Mention that `scipy.interpolate.make_interp_spline` is used for the interpolation?

Reply:
Yes, did that now.